### PR TITLE
Refactor Practitioner Details implementation

### LINKED ISSUE #857: [Client Side] implement the practitioner details endpoint.
**Is your feature request related to a problem? Please describe.**
- After creating a [practitioner details fetch endpoint](https://github.com/opensrp/fhircore/issues/853) we need to get, save & use the data in FHIR core. 

**Describe the solution you'd like**
**Getting data**
- [x] Implement a GET of the [practitioner details fetch endpoint](https://github.com/opensrp/fhircore/issues/853)  as the 1st endpoint call after a user performs a remote login.  
     -  The endpoint parameter should be the user/practitioner id from keycloak.  i.e the keycloak uuid

**Saving data**
- [x] Currently, we have some keycloak response saved in the secure shared preferences. Update this to save all the keycloak info included in the [practitioner details fetch endpoint](https://github.com/opensrp/fhircore/issues/853) response. 
- [x] Save all the resources in the payload mentioned above in the `ResourceEntity` table. 

**Using data**
- [x] Write helper methods to fetch the keycloak related data from the secure shared preferences
      - Should be able to get all demographics
      - Should be able to get all the roles assigned to the user. 
- [x] Write helper a method to fetch all the practitioner-related resources.
      - This can be generic i.e give it the `practitioner-id `and `resource type` then it returns the exact resource required. 

**Additional context**
- [Sample payload](https://github.com/opensrp/fhircore/issues/853)

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #857 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by FikriMilano** [comment]
@dubdabasoduba @f-odhiambo 
This is now blocked by https://github.com/opensrp/fhir-common-utils/issues/5

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/PractitionerDetailsUtil.kt [hunk 1]
```diff
@@ -19,22 +19,31 @@ package org.smartregister.fhircore.engine.util
...
-    fhirContext.registerCustomType(UserBioData::class.java)
+  fun getUrl(keycloakUuid: String) = "$PRACTITIONER_DETAILS_ENDPOINT?$PRACTITIONER_DETAILS_KEYCLOAK_UUID_PARAMETER=$keycloakUuid"
 
-    return fhirContext.newJsonParser()
+  private val fhirR4Cached: FhirContext = FhirContext.forR4Cached()
+
+  fun getCustomJsonParser(): IParser {
+    return fhirR4Cached.apply {
+      registerCustomType(FhirCareTeamExtension::class.java)
+      registerCustomType(FhirOrganizationExtension::class.java)
```
Comment (FikriMilano) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/PractitionerDetailsUtil.kt:41]: @dubdabasoduba 
These classes are missing from the library, even though I already added the imports.

Comment (FikriMilano) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/PractitionerDetailsUtil.kt:41]: Resolved

**Comment by FikriMilano** [comment]
Now unblocked

**Comment by pld** [comment]
@FikriMilano why are we merging this into a branch and not main? (Just wondering, I don't have context for this, thanks)

**Comment by FikriMilano** [comment]
@pld 
Well, it's because I am planning to keep Raazia's credit when we merge the other branch to main.
No technical reason.

**Pull request merged**